Gems/PhysX/Code/Source/Utils.cpp

```diff
@@ -134,52 +192,15 @@ namespace PhysX
                         {
-                            // Our source data is providing one material index per vertex, but PhysX wants one material index
-                            // per triangle.  The heuristic that we'll go with for selecting the material index is to choose
-                            // the material for the vertex that's not on the diagonal of each triangle.
-                            // Ex:  A *---* B
-                            //        | / |      For this, we'll use A for index0 and D for index1.
-                            //      C *---* D
-                            //
-                            // Ex:  A *---* B
-                            //        | \ |      For this, we'll use C for index0 and B for index1.
-                            //      C *---* D
-                            //
-                            // This is a pretty arbitrary choice, so the heuristic might need to be revisited over time if this
-                            // causes incorrect or unpredictable physics material mappings.
-
-                            switch (currentSample.m_quadMeshType)
-                            {
-                            case Physics::QuadMeshType::SubdivideUpperLeftToBottomRight:
-                                currentPhysxSample.materialIndex0 = samples[GetIndex(row + 1, col)].m_materialIndex;
-                                currentPhysxSample.materialIndex1 = samples[GetIndex(row, col + 1)].m_materialIndex;
-                                // Set the tesselation flag to say that we need to go from UL to BR
-                                currentPhysxSample.materialIndex0.setBit();
-                                break;
-                            case Physics::QuadMeshType::SubdivideBottomLeftToUpperRight:
-                                currentPhysxSample.materialIndex0 = currentSample.m_materialIndex;
-                                currentPhysxSample.materialIndex1 = samples[GetIndex(row + 1, col + 1)].m_materialIndex;
-                                break;
-                            case Physics::QuadMeshType::Hole:
-                                currentPhysxSample.materialIndex0 = physx::PxHeightFieldMaterial::eHOLE;
-                                currentPhysxSample.materialIndex1 = physx::PxHeightFieldMaterial::eHOLE;
-                                break;
-                            default:
-                                AZ_Warning("PhysX Heightfield", false, "Unhandled case in CreatePxGeometryFromConfig");
-                                currentPhysxSample.materialIndex0 = 0;
-                                currentPhysxSample.materialIndex1 = 0;
-                                break;
-                            }
+                            // Set the tesselation flag to say that we need to go from UL to BR
+                            currentPhysxSample.setTessFlag();
```

Comment:
By always setting the TessFlag, doesn't this mean that the QuadMeshType that's passed in on each sample is getting ignored?  The previous logic that was here only conditionally set the flag.

Reply:
We only `setTessFlag()` if `(currentSample.m_quadMeshType == Physics::QuadMeshType::SubdivideUpperLeftToBottomRight)`.

See the line 200 above. 

May be a bit difficult to see in the diff mode which code is where. 
This is how it looks:

```
                        if (currentSample.m_quadMeshType == Physics::QuadMeshType::SubdivideUpperLeftToBottomRight)
                        {
                            // Set the tesselation flag to say that we need to go from UL to BR
                            currentPhysxSample.setTessFlag();
                        }

```

Reply:
Ah yep the diff obscured it, ty!